### ISSUE TITLE
Stability and workflow issues

### ISSUE THREAD

**Issue by PYROTEKNIK** [swampservers/contrib#288]
Its understandable that things will obviously break when new features are added, but it's becoming a little bit intimidating to work around it with no information. Would it be asking too much to request some basic information when these bigger, less obvious changes happen? Few of the commits are labeled properly

Certain branches i've been working on haven't been finished, mostly because i feel i am lacking direction. i want to try and address this soon

Evidently, the stability issues are more important to address at the moment. If there is anything I can do to help with that, let me know.

**Comment by swampservers** [comment]
I don't really know what the complaint is; lack of commit messages? I just recently added something that would automatically commit/push changes to make sure everything gets backed up but I still manually commit sometimes and I can do it more often.

It seems to me that most of the recent issues with your additions could have been caught using the client devtools, so I don't see what else I am supposed to do there. 

If you want to fix stability issues, you can start with the client errors in your SWEPs caused by invalid entities (which happens upon joining or upon full refresh, when all the entities get "reset"). Here is one that still exists: 
```
[_swamp] addons/_swamp/lua/weapons/weapon_monke.lua:199: Tried to use a NULL entity!
  1. AnimResetGestureSlot - [C]:-1
   2. y - addons/_swamp/lua/weapons/weapon_monke.lua:199
```

**Comment by PYROTEKNIK** [comment]
sorry, it's just been a bit more difficult lately doing client dev on the server lately. there's been a few different changes that have created debug spam and crashes, broke inventory, or made the server inaccessible. I get why it happens, it's just been hard to predict when it's actually safe to get on and try working on stuff lately

**Comment by swampservers** [comment]
>debug spam

I remove this when I can

>crashes

It's gmod, crashes are very rarely my fault

>broke inventory

Isolated bug to one player who probably has a corrupt item or something

>made the server inaccessible

The server randomly freezes while I am asleep, so you make a weird unclear github issue?

What is the purpose of making this issue? If you need direction pay attention to the things i've suggested for you to make, not make yet more le random outcomes for the secret button that people will barely ever see, which I've already told you several times should be considered done. Things I've already asked for, off the top of my head include:
- Customizer gadgets and cleanup
- Fix hat UVs to fill the space
- Fix ponymodel hitboxes and give it a "head" (maybe just the eyes to make it harder to hit)

I suggest having a notepad on your desk.

**Comment by PYROTEKNIK** [comment]
I made an issue about it because it's been happening a lot more frequently lately, usually after major changes, without warning and there is little that can be done when it happens other than spending extra hours unsuccessfully trying to emulate frequently changing pointshop server behavior from my local copy. All I am asking is that you consider some kind of contingency plan. One example of a solution would be seeing if there's a reasonable way to run the dev server more often without it impacting the main server performance. I know most of the time these issues are completely out of your hands but it makes sense to try and prepare for them anyways

I know exactly what you want from me, and I am doing my best to make time to do them, but I also feel like I should put some time into finishing things I've already started. The customization improvements are something I have been actively working on, it's just requiring a bit more thought than the usual easy stuff, especially given my inexperience working changes into other people's code.

I've started on projects that have been specifically asked for, like the target shooting minigame, but a few of these I have gotten stuck on or unsure about, and I seem to be a bad judge of whether or not contacting you about it is frivolous or not.

I'm really not trying to cause problems, I'm just trying to enjoy what I do for as long as possible. And to me this means sometimes working on some of the less urgent stuff, and not having a completely unmanageable list of projects. I am doing my best to learn from my mistakes, and do projects you ask for, and it still feels like a lot of what I do on my own volition pisses you off.
